daemon/api_prompting.go

```diff
@@ -0,0 +1,381 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2024 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package daemon
+
+import (
+	"encoding/json"
+	"errors"
+	"math"
+	"net/http"
+	"strconv"
+
+	"github.com/snapcore/snapd/interfaces/prompting"
+	"github.com/snapcore/snapd/interfaces/prompting/requestprompts"
+	"github.com/snapcore/snapd/interfaces/prompting/requestrules"
+	"github.com/snapcore/snapd/overlord/auth"
+	"github.com/snapcore/snapd/overlord/ifacestate/apparmorprompting"
+	"github.com/snapcore/snapd/strutil"
+)
+
+var (
+	requestsPromptsCmd = &Command{
+		Path:       "/v2/interfaces/requests/prompts",
+		GET:        getPrompts,
+		ReadAccess: interfaceOpenAccess{Interfaces: []string{"snap-interfaces-requests-control"}},
+	}
+
+	requestsPromptCmd = &Command{
+		Path:        "/v2/interfaces/requests/prompts/{id}",
+		GET:         getPrompt,
+		POST:        postPrompt,
+		ReadAccess:  interfaceOpenAccess{Interfaces: []string{"snap-interfaces-requests-control"}},
+		WriteAccess: interfaceOpenAccess{Interfaces: []string{"snap-interfaces-requests-control"}},
```

Comment:
this also should be interfaceAuthenticatedAccess

Reply:
No, otherwise afaik the user has to authenticate again every time they answer a prompt. Basically, they'll feel like they're getting two prompts for everything they try to do.

For rules, since they result in silently replying to prompts in the future and also require being manually asked for in the first place, we use polkit authentication to make sure the user really wants to do this. But maybe polkit authentication is unnecessary/annoying even for rules?